src/parsers/typescript_parser.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * 
+ * Note (2019/04/16): 
+ * 
+ * This is a direct port from the deprecated jest-test-typescript-parser in https://github.com/facebook/jest, as jest moving to babel-7, which includes direct typescript 
+ * support, the package is removed.
+ * 
+ * However "jest-test-typescript-parser" is used by 3rd party tools such as vscode-jest, together with "jest-editor-support",
+ * therefore, for the continuity and backward compatibility, we merged the legacy parser here and take responsibility in selecting the 
+ * proper parser for the given file. Therefore tools that adopt the new "jest-editor-support" package should remove 
+ * "jest-test-typescript-parser".
+ * 
+ * @flow
+ */
+
+import {readFileSync} from 'fs';
+
+import ts from 'typescript';
+import {
+  NamedBlock,
+  ParsedRange,
+  ParsedNode,
+  ParseResult,
+} from './parser_nodes';
+import type {ParsedNodeType} from './parser_nodes';
+
+export function parse(file: string, data: ?string): ParseResult {
+  const sourceFile = ts.createSourceFile(
+    file,
+    data ? data : readFileSync(file).toString(),
+    ts.ScriptTarget.ES3,
```

Comment:
Is there any reason for not using `ts.ScriptTarget.Latest`?

Reply:
I am not sure why it is ES3... this was just copied from the old repo, probably it hasn't caused any problem so nobody bothered to update it...? 

I am thinking it might be better to just leave it as it is to maintain backward compatibility; and besides, we might be moving away from this parser in favor of babel's typescript parser going forward...

Reply:
I think it doesn't matter at all since the code file doesn't get transpiled but we only obtain the AST.